frontend/app/mocks/subscription-api.server.ts

```diff
@@ -0,0 +1,91 @@
+import { HttpResponse, http } from 'msw';
+import { z } from 'zod';
+
+import { db } from './db';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('subscription-api.server');
+
+const subscriptionApiSchema = z.object({
+  id: z.string(),
+  sin: z.string(),
+  email: z.string(),
+  subscribed: z.boolean(),
+  preferredLanguage: z.string(),
+});
+
+/**
+ * Server-side MSW mocks for the subscription API.
+ */
+export function getSubscriptionApiMockHandlers() {
+  log.info('Initializing Subscription API mock handlers');
+
+  return [
+    //
+    // Handler for GET request to retrieve email alerts decription by sin
+    //
+    http.get('https://api.example.com/v1/subscriptions/:userId', ({ params, request }) => {
+      log.debug('Handling request for [%s]', request.url);
+
+      const url = new URL(request.url);
+      const alertType = url.searchParams.get('alertType');
```

Comment:
No need to read a query param. Depending on whatever `userId` is passed, the corresponding alert type will be returned for that `userId`.